.github/workflows/release_docker.yml

```diff
@@ -0,0 +1,114 @@
+# **what?**
+# This workflow will generate a series of docker images for dbt and push them to the github container registry
+
+# **why?**
+# Docker images for dbt are used in a number of important places throughout the dbt ecosystem.  This is how we keep those images up-to-date.
+
+# **when?**
+# This is triggered manually 
+
+# **TODO**
+# - build this into the release workflow (or conversly, break out the different release methods into their own workflow files)
+# - find a workaround for *&^%&*^ing dbt-postgres (currently not supported)
+
+name: Docker release
+
+on:
+  workflow_dispatch:
+    inputs:
+      package:
+       description: The package to release. _One_ of [dbt-core, dbt-redshift, dbt-bigquery, dbt-snowflake, dbt-spark]
+       required: true
+      version_number:
+       description: The release version number (i.e. 1.0.0b1). Do not include `latest` tags or a leading `v`!
+       required: true
+
+jobs:
+  get_version_meta:
+    name: Get version meta
+    runs-on: ubuntu-latest
+    outputs:
+      major: ${{ steps.version.outputs.major }}
+      minor: ${{ steps.version.outputs.minor }}
+      patch: ${{ steps.version.outputs.patch }}
+      latest: ${{ steps.latest.outputs.latest }}
+      minor_latest: ${{ steps.latest.outputs.minor_latest }}
+    steps:
+      - uses: actions/checkout@v1
+      - name: Split version
+        id: version
+        run: |
+          IFS="." read -r MAJOR MINOR PATCH <<< ${{ github.event.inputs.version_number }}
+          echo "::set-output name=major::$MAJOR"
+          echo "::set-output name=minor::$MINOR"
+          echo "::set-output name=patch::$PATCH"
+
+      - name: Is pkg 'latest'
+        id: latest
+        uses: ./.github/actions/latest-wrangler
+        with:
+          package: ${{ github.event.inputs.package }}
+          new_version: ${{ github.event.inputs.version_number }}
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+          halt_on_missing: False
+
+  setup_image_builder:
+    name: Set up docker image builder
+    runs-on: ubuntu-latest
+    needs: [get_version_meta]
+    steps:      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+  build_and_push:
+    name: Build images and push to GHCR
+    runs-on: ubuntu-latest
+    needs: [setup_image_builder, get_version_meta]
+    steps:
+      - name: Get docker build arg
+        id: build_arg
+        run: |
+          echo "::set-output name=build_arg_name::"$(echo ${{ github.event.inputs.package }} | sed 's/\-/_/g')
+          echo "::set-output name=build_arg_value::"$(echo ${{ github.event.inputs.package }} | sed 's/postgres/core/g')
+
+      - name: Log in to the GHCR
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push PATCH
```

Comment:
The name of this step is slightly confusing to me with the word "Patch". This is building and pushing the image with just with the given version number as the tag, correct? Basically what we would do if latest wasn't involved?

Reply:
Yeah.  I'll make this more clear.  I refer to it as pushing the PATCH image because it's tag will be the full semver all the way down to the PATCH level.
`MAJOR.MINOR.PATCH`